parley/src/tests/test_basic.rs

```diff
@@ -112,6 +112,46 @@ fn full_width_inbox() {
     }
 }
 
+#[test]
+fn inboxes_separated_by_whitespace() {
+    let mut env = testenv!();
+
+    let text = "   ";
+    let mut builder = env.ranged_builder(text);
```

Comment:
Shouldn't this test use `TreeBuilder`? It currently has the exact same behavior on main, so it doesn't really demonstrate the bug this PR is fixing.

Reply:
Yes, yes it should. The new snapshot has some grey rectangles around the inline boxes. What do they represent?